share/revng/test/configuration/revng/scripting.yml

```diff
@@ -0,0 +1,171 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+commands:
+  #
+  # Test python scripting cli
+  #
+  - type: revng.scripting-cli
+    from:
+      - type: revng-qa.compiled
+        filter: example-executable-1 and with-debug-info
+    command: |-
+      python scripting-cli.py "$OUTPUT" "$INPUT"
+    scripts:
+      scripting-cli.py: |-
+        #!/usr/bin/env python3
+
+        from revng.scripting import CLIProject
+        from revng import ptml
+        import re
+        import sys
+
+        # Create cli project, import binary and run analysis lists
+        project = CLIProject(sys.argv[1])
+        project.import_binary(sys.argv[1], sys.argv[2])
+        project.analyze("revng-initial-auto-analysis")
+        project.analyze("revng-c-initial-auto-analysis")
+
+        # Get the first function with `CustomName` and the functions index
+        function_original_name = ""
+        function_idx = 0
+        for idx, function in enumerate(project.model.Functions):
+            if function.CustomName:
+                function_original_name = function.CustomName
+                function_idx = idx
+                break
+        assert function_original_name != ""
+
+        # Get artifact by using the project and function, the result should be the same
+        result1 = project.model.Functions[function_idx].fetch("disassemble")
+        result2 = project.fetch("disassemble", [project.model.Functions[function_idx]])
+        assert ptml.parse(result1) == ptml.parse(result2)
+
+        # Make sure that we get the correct function name in the parse result
+        re_match = re.search(function_original_name, ptml.parse(result1))
+        assert re_match is not None
+
+        # Change the function name, get an artifact and make sure that the 
+        # parsed result contains the new name
+        function_new_name1 = "lol"
+        project.model.Functions[function_idx].CustomName = function_new_name1
+        result1 = project.model.Functions[function_idx].fetch("disassemble")
+        re_match = re.search(function_new_name1, ptml.parse(result1))
+        assert re_match is not None
+
+        # Change function name again and save
+        function_new_name2 = "lolz"
+        project.model.Functions[function_idx].CustomName = function_new_name2
+        project.save()
+
+        # Load the project again and run the analysis again
+        project.import_binary(sys.argv[1], sys.argv[2])
+        project.analyze("revng-initial-auto-analysis")
+        project.analyze("revng-c-initial-auto-analysis")
```

Comment:
There should be a shorthand, like `project.import_and_analyze` that does these 3 things